src/app/backend/events.go

```diff
@@ -164,6 +166,26 @@ func GetReplicaSetPodsEvents(client *client.Client, namespace, replicaSetName st
 	return events, nil
 }
 
+// Gets events associated to given pod
+func GetPodEvents(client *client.Client, pod api.Pod) (*api.EventList, error) {
+	fieldSelector, err := fields.ParseSelector("involvedObject.name=" + pod.Name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	list, err := client.Events(pod.Namespace).List(unversioned.ListOptions{
```

Comment:
There are N api calls even now. I've only refactored it and extracted function.

https://github.com/kubernetes/dashboard/blob/master/src/app/backend/events.go#L149